src/gui/syncconflictsmodel.cpp

```diff
@@ -0,0 +1,355 @@
+            mConflictData.push_back({});
+            continue;
+        }
+
+        const auto conflictedRelativePath = oneConflict._file;
+        const auto dbRecord = folder->journalDb();
+        const auto baseRelativePath = dbRecord ? dbRecord->conflictFileBaseName(conflictedRelativePath.toUtf8()) : QString{};
+
+        const auto dir = QDir(folder->path());
+        const auto conflictedPath = dir.filePath(conflictedRelativePath);
+        const auto basePath = dir.filePath(baseRelativePath);
+
+        const auto existingFileInfo = QFileInfo(basePath);
+        const auto conflictFileInfo = QFileInfo(conflictedPath);
+
+        auto newConflictData = ConflictInfo{
+            existingFileInfo.fileName(),
+            mLocale.formattedDataSize(existingFileInfo.size()),
+            mLocale.formattedDataSize(conflictFileInfo.size()),
+            existingFileInfo.lastModified().toString(),
+            conflictFileInfo.lastModified().toString(),
+            QUrl{QStringLiteral("image://tray-image-provider/:/fileicon") + existingFileInfo.filePath()},
+            QUrl{QStringLiteral("image://tray-image-provider/:/fileicon") + conflictFileInfo.filePath()},
+            false,
+            false,
+            existingFileInfo.filePath(),
+            conflictFileInfo.filePath(),
+        };
+
+        mConflictData.push_back(std::move(newConflictData));
+    }
+}
+
+void SyncConflictsModel::setExistingSelected(bool value,
+                                             const QModelIndex &index,
+                                             int role)
+{
+    mConflictData[index.row()].mExistingSelected = value;
+    Q_EMIT dataChanged(index, index, {role});
+
+    if (!mConflictData[index.row()].mExistingSelected && mAllExistingsSelected) {
+        mAllExistingsSelected = false;
+        Q_EMIT allExistingsSelectedChanged();
+    } else if (mConflictData[index.row()].mExistingSelected && !mAllExistingsSelected) {
+        auto allSelected = true;
+        for (const auto &singleConflict : qAsConst(mConflictData)) {
+            if (!singleConflict.mExistingSelected) {
+                allSelected = false;
+                break;
+            }
+        }
+        if (allSelected) {
+            mAllExistingsSelected = true;
+            Q_EMIT allExistingsSelectedChanged();
+        }
+    }
+}
+
+void SyncConflictsModel::setConflictingSelected(bool value,
+                                                const QModelIndex &index,
+                                                int role)
+{
+    mConflictData[index.row()].mConflictSelected = value;
+    Q_EMIT dataChanged(index, index, {role});
+
+    if (!mConflictData[index.row()].mConflictSelected && mAllConflictingsSelected) {
+        mAllConflictingsSelected = false;
+        Q_EMIT allConflictingSelectedChanged();
+    } else if (mConflictData[index.row()].mConflictSelected && !mAllConflictingsSelected) {
+        auto allSelected = true;
```

Comment:
@mgallien you may wanna use `std::find_if` here and make `const auto allSelected = std::find_if(...) != std::end(mConflictData)` shorter

Reply:
done